api/src/main/java/org/openmrs/module/fhir2/providers/r4/PractitionerFhirResourceProvider.java

```diff
@@ -59,24 +60,16 @@ public Practitioner getPractitionerById(@IdParam @NotNull IdType id) {
 	@History
 	@SuppressWarnings("unused")
 	public List<Resource> getPractitionerHistoryById(@IdParam @NotNull IdType id) {
-		Practitioner practitioner = practitionerService.getPractitionerByUuid(id.getIdPart());
+		Practitioner practitioner = practitionerService.get(id.getIdPart());
 		if (practitioner == null) {
 			throw new ResourceNotFoundException("Could not find practitioner with Id " + id.getIdPart());
 		}
 		return practitioner.getContained();
 	}
 	
 	@Search
-	@SuppressWarnings("unused")
-	public Bundle findPractitionersByName(@RequiredParam(name = Practitioner.SP_NAME) @NotNull String name) {
-		return FhirProviderUtils.convertSearchResultsToBundle(practitionerService.findPractitionerByName(name));
-	}
-	
-	@Search
-	@SuppressWarnings("unused")
-	public Bundle findPractitionersByIdentifier(
-	        @RequiredParam(name = Practitioner.SP_IDENTIFIER) @NotNull String identifier) {
-		return FhirProviderUtils.convertSearchResultsToBundle(practitionerService.findPractitionerByIdentifier(identifier));
+	public IBundleProvider searchForPractitioners(@OptionalParam(name = Practitioner.SP_NAME) StringAndListParam name,
```

Comment:
So we should also add support for searching by `given` and `family` as well as searching by `address`, `address-city`, `address-country`, `address-postalcode`, and `address-state`.

Reply:
@ibacher Can i do that as a separate ticket? So that we atleast unblock the paging part.

Reply:
Ok